lua/weapons/remotecontroller.lua

```diff
@@ -41,40 +41,55 @@ function SWEP:Holster()
 	if (self.Linked) then
 		self:Off()
 	end
+	
 	return true
 end
 
 function SWEP:OnDrop()
-	if (self.Linked) then
-		self:Off()
-		self.Linked = nil
-	end
+	if (!self.Linked) then return end
```

Comment:
Good change, but can you please separate cleanup and actual changes into two commits?
Also, please use standard Lua syntax.
Those () around the if condition are also not necessary.